### PR TITLE
extend coverage for  e2e tests for plugin cli options

### LINKED ISSUE #1760: 🚧 Dev ➾ extend e2e test coverage of plugins to include all options
### 🔗 Epic|Feature Link

_No response_

### 🚥 Definition of Done

- [ ] 🎯 Planning and Design
- [ ] 🏎️ Dev Implementation Complete
- [ ] ♾️ PR Merged
- [ ] 🏆 Product Acceptance

### 🧰 Work Description

Extend e2e test coverage of plugins to include all options

### ✅ To Do

Extend e2e test coverage of plugins to include all options

### ⚖️ Acceptance Criteria

✨ Functional: 
Extend e2e test coverage of plugins to include all options

### 🔭 Scope

🙅‍♂️ Out of Scope: 
- ... 
 
 other plugins

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by michaelkernaghan**
# 🌮 Taqueria PR
closes #1760
## 🪁 Description

extend coverage for  e2e tests for plugin cli options

## 🪂 Pre-Merge Checklist (Definition of Done)

🚦 Required to merge:

- [x ] ⛱️ I have completed this PR template in full and updated the title appropriately
- [x ] ⛵ My code builds cleanly, and I have manually tested the changes
- [ ] 🏄‍♂️ Another team member has built this branch and done manual testing on the change
- [x ] 🏖️ New and existing unit tests pass locally and in CI
- [ ] 🔱 The test plan has been implemented and verified by an SDET
- [x ] 🦀 Automated tests have been written and added to this PR
- [x ] 🐬 I have commented my code, particularly in hard-to-understand areas
- [x ] 🤿 Corresponding changes have been made to all documentation
- [x ] 🐚 Required changes to the VScE have been made
- [x ] 🪸 Required updates to scaffolds have been made
- [ ] 🚢 The release checklist has been completed

## 🛩️ Summary of Changes

extend coverage for  e2e tests for plugin cli options

## 🎢 Test Plan

tests keep working

## 🛸 Type of Change

- [x] 🧽 Chore ➾
- [ ] 🛠️ Fix ➾
- [ ] ✨ Feature ➾
- [ ] 👷 Refactor ➾
- [ ] 🧪 Pre-Release ➾
- [ ] 🚀 Release ➾

### CODE REVIEW

**Review by egarson — APPROVED** [review]
LGTM

File: tests/e2e/archetype-plugin-e2e-tests.spec.ts [hunk 1]
```diff
@@ -6,24 +6,25 @@ import { prepareEnvironment } from '@gmrchk/cli-testing-library';
...
 		await exists('./test-project/.taq/config.json');
-		const {} = await execute('taq', 'install ../taqueria-plugin-archetype', './test-project');
+		await execute('taq', 'install ../taqueria-plugin-archetype', './test-project');
 		await exists('./test-project/node_modules/@taqueria/plugin-archetype/index.js');
 
 		const { stdout } = await execute('taq', 'compile --help', './test-project');
 		expect(stdout).toEqual(
 			expect.arrayContaining(['Compile a smart contract written in a Archetype syntax to Michelson code']),
 		);
+		console.log(stdout);
```
Comment (mweichert) [inline tests/e2e/archetype-plugin-e2e-tests.spec.ts:18]: Was this console.log for debugging purposes? Should it be removed or commented out?

File: tests/e2e/archetype-plugin-e2e-tests.spec.ts [hunk 2]
```diff
@@ -32,14 +33,31 @@ describe('Archetype Plugin E2E Testing for Taqueria CLI', () => {
...
+	test('archetype plugin will support cli options', async () => {
+		const { execute, cleanup, spawn } = await prepareEnvironment();
+		const { waitForText } = await spawn('taq', 'init test-project --debug');
+		await waitForText("Project taq'ified!");
+
+		const { stdout } = await execute('taq', 'install ../taqueria-plugin-archetype', './test-project');
+		expect(stdout).toContain('Plugin installed successfully');
+
+		const { stdout: stdout2 } = await execute('taq', '--version', './test-project');
+		expect(stdout2).not.toEqual([]);
```
Comment (mweichert) [inline tests/e2e/archetype-plugin-e2e-tests.spec.ts:45]: Should we also assert that nothing was output to stderr?

File: tests/e2e/ligo-plugin-e2e-tests.spec.ts [hunk 3]
```diff
@@ -55,6 +55,32 @@ describe('Ligo Plugin E2E Testing for Taqueria CLI', () => {
...
+	test('ligo plugin will support cli options', async () => {
+		const { execute, cleanup, writeFile, readFile, ls, spawn } = await prepareEnvironment();
+		const { waitForText } = await spawn('taq', 'init test-project --debug');
+		await waitForText("Project taq'ified!");
+
+		const { stdout } = await execute('taq', 'install ../taqueria-plugin-ligo', './test-project');
+		expect(stdout).toContain('Plugin installed successfully');
+
+		const { stdout: stdout2 } = await execute('taq', '--version', './test-project');
+		expect(stdout2).not.toEqual([]);
```
Comment (mweichert) [inline tests/e2e/ligo-plugin-e2e-tests.spec.ts:67]: Should we also assert that nothing was output to stderr?

File: tests/e2e/ligo-plugin-e2e-tests.spec.ts [hunk 4]
```diff
@@ -55,6 +55,32 @@ describe('Ligo Plugin E2E Testing for Taqueria CLI', () => {
...
+
+		const { stdout: stdout3 } = await execute('taq', '--build', './test-project');
+		expect(stdout3).not.toEqual([]);
```
Comment (mweichert) [inline tests/e2e/ligo-plugin-e2e-tests.spec.ts:70]: Should we also assert that nothing was output to stderr?

**Review by mweichert — APPROVED** [review]
Looks good to me, although I did have some questions which may or may not require some adjustments.

File: tests/e2e/archetype-plugin-e2e-tests.spec.ts (hunk 1 above)
Comment (michaelkernaghan) [reply tests/e2e/archetype-plugin-e2e-tests.spec.ts:18]: thanks, should be removed. will remove it.

File: tests/e2e/archetype-plugin-e2e-tests.spec.ts (hunk 2 above)
Comment (michaelkernaghan) [reply tests/e2e/archetype-plugin-e2e-tests.spec.ts:45]: i can, but it implied

File: tests/e2e/archetype-plugin-e2e-tests.spec.ts [hunk 5]
```diff
@@ -32,14 +33,31 @@ describe('Archetype Plugin E2E Testing for Taqueria CLI', () => {
...
+
+		const { stdout: stdout3 } = await execute('taq', '--build', './test-project');
+		expect(stdout3).not.toEqual([]);
```
[the remark this answers is not in the document]
Comment (michaelkernaghan) [reply tests/e2e/archetype-plugin-e2e-tests.spec.ts:48]: for the command options cases, maybe it's a good idea to be explicit. thanks, I will add it

File: tests/e2e/ligo-plugin-e2e-tests.spec.ts (hunk 3 above)
Comment (michaelkernaghan) [reply tests/e2e/ligo-plugin-e2e-tests.spec.ts:67]: added

File: tests/e2e/ligo-plugin-e2e-tests.spec.ts (hunk 4 above)
Comment (michaelkernaghan) [reply tests/e2e/ligo-plugin-e2e-tests.spec.ts:70]: added

**Pull request merged**